src/graphnet/data/sqlite_dataconverter.py

```diff
@@ -62,6 +62,8 @@ def __init__(
              "I3TruthExtractor to allow for attaching unique indices.")
 
         self._table_names = [extractor.name for extractor in self._extractors]
+        self._pulsemap = [x for x in self._table_names if is_pulsemap_check(x)][0] #the first non-truth non-retro table name (pulsemap)
```

Comment:
1. This may introduce a level of ambiguity in cases where there are multiple non-truth, non-retro pulsemaps, e.g., as may be the case in Upgrade data.
2. If we want to get a list of the (one or more) non-truth, non-retro pulsemap(s), I also think that it might be better to do `self._pulsemaps = [extractor.name for extractor in self._extractors if isinstance(extractor, I3FeatureExtractor)]` rather than doing string-based matching, as all feature extraction inherits from this class, whereas `I3RetroExtractor` and `I3TruthExtractor` don't.